atmo/users/adapters.py

```diff
@@ -33,3 +35,11 @@ def validate_disconnect(self, account, accounts):
             raise forms.ValidationError(
                 'At least one account needs to be connected'
             )
+
+    def populate_user(self, request, sociallogin, data):
+        user = super().populate_user(request, sociallogin, data)
+        email = user_email(user)
+        if email and '@' in email:
+            username = email.split('@')[0]
+            user_username(user, username)
+        return user
```

Comment:
Not sure where or how this is used. Do you need to do a `user.save()` here since you're setting the username attribute?